[비동기 댓글 기능] 이바 미션 제출합니다.

seonghun127 opened the pull request:
안녕하세요 브라운! 리뷰이 이바입니다.

**Lv2 들어서 항상 좋은 수업, 감사합니다 ㅎㅎㅎ**

이번 미션을 진행하면서 AJAX에 대해 간략하게나마 알 수 있었습니다.
사실 명확하게 AJAX가 뭐다라고 말할 수 없을거 같긴 하지만요...

이번에도 부족한 부분에 대해서 많은 지적 부탁드리겠습니다.

seonghun127 committed:
setting: mission complete

seonghun127 committed:
docs: README.md

1. todo 목록 정리

seonghun127 committed:
feat: build.gradle

1. devtools 의존성 추가

seonghun127 committed:
feat: ArticleController Class

1. Article 도메인 객체 구현
2. Test 추가
3. ArticleRepository 저장기능 구현

seonghun127 committed:
feat: layout

1. build.gradle 의존성 추가
2. 공통 레이아웃 추출 및 적용

seonghun127 committed:
feat: ArticleDto Class

1. Create 용도의 ArticleDto 구현
2. ArticleDto를 사용하도록 Controller와 Repository 변경

seonghun127 committed:
refactor: Article class

1. ArticleDto를 사용하기 위한 생성자 추가
2. id 자동 증가를 위한 static field 추가

seonghun127 committed:
feat: ArticleController class

1. 조회 기능 구현

seonghun127 committed:
style: css 및 templates 파일 코드 포메팅

seonghun127 committed:
fix: app.css

1. 파일 오류 수정

seonghun127 committed:
test: ArticleRepository class

1. 조회 및 전체조회 테스트

seonghun127 committed:
fix: layout

1. thymeleaf-layout 버전에 따른 오류 수정

seonghun127 committed:
fix: ArticleController class

1. createArticle 리다이렉팅 주소 수정

seonghun127 committed:
feat: ArticleController class

1. 게시물 수정 기능 구현
2. ArticleDto.Update 생성
3. ArticleRepository에 update 기능 추가
4. update를 위해 Article 도메인 객체에 생성자 추가

seonghun127 committed:
feat: article-edit.html

1. 생성 화면과 수정 화면에 따른 form 경로 및 메서드 조정
2. article.html에 하드코딩된 url 변경

seonghun127 committed:
feat: ArticleController class

1. 삭제 기능 구현
2. ArticleRepository delete 구현

seonghun127 committed:
feat: article.html

1. delete 버튼 구현

seonghun127 committed:
refactor: layout

1. header 공통부분 추출

seonghun127 committed:
fix: ArticleRepository class

1. 수정시 새 게시물이 추가되는 오류 수정

seonghun127 committed:
style: 전체 코드 포매팅

seonghun127 committed:
fix: article-edit.html

1. 수정 페이지에서 글 내용이 나오지 않는 오류 수정

seonghun127 committed:
fix: article.html

1. 예시로 설정된 게시글 내용 삭제
2. 작성된 게시글 내용이 나오게끔 수정

seonghun127 committed:
refactor: custom head 적용 방법 변경

seonghun127 committed:
refactor: fragment 폴더 생성

seonghun127 committed:
refactor: layout.html

1. main-head를 담는 th:replace 태그 추가
2. main-body를 담는 th:replace 태그 추가

seonghun127 committed:
refactor: ArticleController class

1. 필드 의존 주입에서 생성자 의존 주입으로 변경

seonghun127 committed:
refactor: NotFoundArticleException class 추가

seonghun127 committed:
refactor: ArticleRepository class

1. find() 메서드의 이름 수정 : findById()

seonghun127 committed:
style: 코드 포매팅

seonghun127 committed:
refactor: Article class

1. set 메서드 제거

seonghun127 committed:
fix: ArticleRepositoryTest 클래스에서 깨지는 테스트 코드 수정

seonghun127 committed:
refactor: ArticleController class

1. @AllArgsContructor 를 @RequiredArgsContructor 어노테이션으로 변경
2. articleRepository 필드를 final 변경
- 불변셩을 유지하기 위해

seonghun127 committed:
feat: Article class

1. 커스터마이즈한 Builder 패턴 추가

seonghun127 committed:
refactor: Article class에 lombok @Builder 적용

seonghun127 committed:
feat: ArticleController class

1. 게시물 추가 테스트 코드 구현

seonghun127 committed:
style: ArticleRepositoryTests class 이름 변경

seonghun127 committed:
fix: Article class

1. updateArticle() 메서드 추가

seonghun127 committed:
refactor: 중복 head 부분 제거

seonghun127 committed:
feat: ArticleController class

1. modelMapper 의존성 주입
2. modelMapper 클래스를 빈으로 등록
3. ArticleDto 클래스에 Response 중첩클래스 추가
4. ArticleController에서 article(s) 를 반환하는 모든 API의 반환 형태를 ArticleDto.Response 타입으로 변경

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
refactor: ArticleController class

1. updateArticle() 메서드 리턴 방식 변경
- 리다이렉트에서 포워딩 방식으로 변경한다
2. 그에 따른 인수 테스트 방식 수정
- 수정된 값을 확인하는 테스트로 수정한다

seonghun127 committed:
style: 주석 제거

seonghun127 committed:
refactor: ArticleRespository class

1. save() 메서드 리턴 타입 void로 변경
2. update() 메서드 리턴 타입 void로 변경

seonghun127 committed:
refactor: ArticleController class

1. createArticle() 메서드 리턴 타입 변경
- 리다이렉트 방식에서 포워딩 방식으로 변경한다

seonghun127 committed:
refactor: ArticleControllerTests class

1. 지속 테스트 가능한 코드로 리펙토링

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
fix: ArticleController class

1. createArticle(), updateArticle() api 리턴 타입 변경
- 포워드 방식에서 리다이렉트 방식으로 변경한다

2. ArticleRepository 클래스에서 save(), update() 메서드의 리턴 타입 변경
- void에서 long으로 변경한다

3. 테스트 코드 수정

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
fix: Spring Data Jpa 사용

seonghun127 committed:
docs: 기능 구현 목록 정리

seonghun127 committed:
docs: h2 기본 설정

seonghun127 committed:
docs: 기능 구현 목록 정리

seonghun127 committed:
feat: 회원가입 요청 페이지 이동 및 저장

1. 회원 가입 페이지 이동 api(renderSignUpPage())를 구현한다
2. 회원 정보를 저장하는 api(createUser())를 구현한다

seonghun127 committed:
merge local branch into fork repository branch

seonghun127 committed:
feat: 회원등록 유효성 검사

seonghun127 committed:
fix: 회원가입 api response 방식 변경

1. createUser() api의 리턴 방식을 리다이렉트로 변경한다
- 회원가입이 성공하면 "/login"으로 리다이렉트 된다
- 회원가입이 실패하면 "/signup"으로 리다이렉트 된다

seonghun127 committed:
feat: 회원 조회 기능 구현

1. 회원을 조회하는 api(readUsers())를 구현한다
2. service 계층에서 dto와 domain 객체의 mapping 처리한다

seonghun127 committed:
fix: 페이지 링크 수정

seonghun127 committed:
feat: 서버에 예외처리 추가

1. email, password, name에 형식 조건을 추가한다
2. ControllerAdvice로 exception handler를 관리한다

seonghun127 committed:
feat: UserControllerTest class

seonghun127 committed:
docs: 기능 구현 목록 정리

seonghun127 committed:
refactor: 서버에서 예외처리 리팩토링

1. 커스텀 예외를 추가한다
- 중복, 형식(이메일, 이름, 비밀번호)
2. ControllerAdvice와 ExceptionHandler를 적용한다
3. Response Status를 설정한다

seonghun127 committed:
feat: 로그인, 로그아웃 기능 구현

1. Login 전용 DTO를 추가한다 (UserDTO.Login)
2. 로그인 성공 시 Session을 생성한다
3. 로그인 유무에 따라 페이지를 분기 이동한다

seonghun127 committed:
fix: 페이지 요청 방식 수정

1. get 요청 시 화면, url 일관성 주기 위해 redirect방식으로 변경한다

seonghun127 committed:
docs: 기능 구현 목록 정리

seonghun127 committed:
feat: MyPage 이동 기능 구현

1. 마이페이지로 이동하는 api(renderMYPage())를 구현한다
2. 로그인 유무에 따라 이동 페이지를 결정한다

seonghun127 committed:
fix: 로그인이 안되있을 시 로그아웃 기능 수정

1. session 정보가 존재할때만 로그아웃을 수행한다

seonghun127 committed:
docs: 기능 목룩 구현 정리

seonghun127 committed:
feat: 회원 정보 수정 기능 추가

1. 회원 정보 수정 api(updateUser())를 추가한다
2. My Page로 이동하는 api(renderMypage())를 수정한다
- session 정보를 확인하는 로직을 삭제한다
3. Edit My Page로 이동하는 api(renderEditMypage())를 추가한다
- renderMypage()에서 처리하던 session 관리 로직을 추가한다
4. UserDto.Reponse 클래스에 id 필드를 추가한다
5. Service 계층에 update() 메서드를 추가한다

seonghun127 committed:
fix: 회원가입, 회원수정 시 예외 처리 코드 분류

1. InvalidSignUpFormException을 구현한다
2. InvalidEditFormException을 구현한다

seonghun127 committed:
feat: 회원 탈퇴 기능 구현

1. 회원 정보를 삭제하는 api(deleteUser()) 추가한다
2. 테스트 코드 @AfterEach에 삭제 api를 호출하여 전체 테스트가 가능하게 한다

seonghun127 committed:
feat: UserService 테스트 코드 추가

seonghun127 committed:
fix: UserService 테스트 코드 수정

seonghun127 committed:
refactor: ArticleService 계층 분리

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
docs: data.sql 추가

seonghun127 committed:
fix: 테스트 코드 오류 수정

1. data.sql 파일 삭제

seonghun127 committed:
refactor: 도메인 @Builder 어노테이션 적용 위치 수정

1. class 전체에 적용하던 @Builder 어노테이션을 생성자로 변경한다
- 생성자 파라미터에 대해서만 빌더 패턴을 적용하기 위해

seonghun127 committed:
refactor: 회원 삭제 시 logout api 호출로 로직 변경

seonghun127 committed:
refactor: 도메인별 패키지 분리

seonghun127 committed:
refactor: 로그인 여부에 따른 이동경로 분기 설정

1. 화면 우측 상단에 이름 / Login 버튼에 대해 로그인 여부에 따라
이동하는 경로를 달리 지정한다
- 로그인 시 '/', 아닐 시 '/login'으로 이동한다

seonghun127 committed:
feat: Interceptor 적용

seonghun127 committed:
feat: Argument Resolver 구현

seonghun127 committed:
refactor: session 정보 삭제 방법 수정

1. invalidate() 메서드를 removeAttribute() 메서드로 변경한다

seonghun127 committed:
refactor: dto 클래스의 중첩 클래스 이름을 명사형으로 수정

seonghun127 committed:
style: 코드 포멧팅

seonghun127 committed:
feat: UserDto.Login 클래스에 validate 추가

1. 이메일과 비밀번호의 형식을 체크한다
2. InvalidLoginFormException을 추가한다

seonghun127 committed:
feat: User 클래스의 필드 validation 추가

seonghun127 committed:
refactor: update 수정

1. 변경할 객체를 생성하고 save()를 다시 호출했던 방식에서
변경할 객체에 update() 메세지를 보내어 수정되는
사항이 영속적으로 저장되는 방식으로 변경한다

seonghun127 committed:
style: import 패키지 변경

seonghun127 committed:
refactor: 로그인시 비밀번호 일치 검사 로직 변경

1. User 클래스에 checkPassword() 메서드를 추가한다
2. User 객체에게 checkPassword() 메세지를 보내 비밀번호를 확인한다

seonghun127 committed:
refactor: 테스트 시 회원가입과 로그인 로직을 따로 추출

1. test code에서 setUp()에서 처리했던 회원가입, 로그인하는 로직을
UserUtilForTest 클래스로 추출한다

seonghun127 committed:
feat: Attribute Converter 구현

1. User 도메인 필드의 타입에 대한 래핑 클래스를 생성한다
2. 서버에선 User 도메인 필드 타입을 모드 래핑 클래스로 적용한다
3. DB에 저장할 시 String 타입으로 저장해주는 래핑 클래스에 대한 Converter를 생성한다

seonghun127 committed:
refactor: Name 클래스의 updateName 메서드 수정

1. VO인 Name 클래스의 불변을 위해 updateName() 메서드 로직을
변경될 필드 값에 대해 새로운 Name 객체로 반환하는 구조로 변경한다

seonghun127 committed:
refactor: VO 클래스 필드에 final 추가

seonghun127 committed:
refactor: domain 패키지 안에 vo 패키지 추가

seonghun127 committed:
refactor: 회원 정보 수정 성공 테스트 리펙토링

1. 정보 수정 요청 후 리다이렉트 되는 페이지에서 수정된 정보가
잘 반영됐는지 확인하는 로직을 추가한다

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
Merge remote-tracking branch 'upstream/seonghun127' into seonghun127

seonghun127 committed:
docs: 기능 목록 구현 업데이트

seonghun127 committed:
feat: 게시글 회원 정보 연결

1. 다대일 매핑

seonghun127 committed:
feat: 게시글 회원 정보 연결

1. 수정, 삭제 시 회원 정보 확인

seonghun127 committed:
feat: 게시글 회원 정보 연결

1. 프론트에서 프로필을 회원 정보에 맞게 변경

seonghun127 committed:
feat: 댓글 생성 기능 추가

1. User 도메인과 Article 도메인을 연결한다

seonghun127 committed:
refactor: Article 클래스의 메서드명 변경

1. matchAuthorId() 에서 notMatchAuthorId()로 변경한다

seonghun127 committed:
feat: 댓글 조회, 수정, 삭제 기능 추가

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
refactor: @Data 어노테이션 추가

seonghun127 committed:
feat: 댓글 생성, 조회, 삭제 기능 구현

seonghun127 committed:
feat: 댓글 수정 기능 구현

seonghun127 committed:
feat: 테스트 코드 구현

seonghun127 committed:
feat: 오류 메세지 출력

seonghun127 committed:
refactor: RedirectView 클래스 사용

1. ArticleController에서 redirect되는 api의 리턴 타입을 RedirectView로 변경한다

seonghun127 committed:
refactor: forward 되는 api의 리턴 값에서 '/' 제거

seonghun127 committed:
refactor: 세션 정보 확인 로직 수정

1. UserSession의 id 값과 uri 쿼리를 비교하는 로직으로 변경한다

seonghun127 committed:
refactor: 테스트 코드 중복 제거

seonghun127 committed:
refactor: ServiceTest 오류 수정

seonghun127 committed:
refactor: 예외 클래스에 정확한 로그 기록을 위해 생성자에 인자 추가

1. User, Article, Comment를 찾을 수 없는 예외 발생 시 해당 객체의 id 값을
예외 클래스에 넘겨주어 id까지 예외 메세지로 출력한다

seonghun127 committed:
refactor: dto 분리

1. 도메인별 DTO 클래스에 중첩 클래스를 사용했던 로직을
따로 분리한다

seonghun127 committed:
refactor: @EqualsAndHashCode 어노테이션 속성 설정

1. Immutable field인 id값으로만 equals, hashcode를 구현하게 한다
- 모든 필드 비교시 발생하는 문제를 방지하기 위함

seonghun127 committed:
refactor: Article, Comment Service 레이어의 중복 코드 제거

1. update, delete 메서드 수행 시 본인인지 확인하는 로직을
checkAuthority() 메서드로 분리한다

seonghun127 committed:
refactor: fragment/header.html을 index.html에서도 적용

seonghun127 committed:
refactor: User Service에서 deleteById() 메서드에 try-catch 문 추가

1. JPA 기본 메서드인 deleteById() 메서드에서 던저지는 오류를 잡고
커스텀으로 만든 NotFoundUserException을 던진다

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
refactor: @RequiredArgsConstructor 어노테이션 삭제

seonghun127 committed:
refactor: User 도메인 클래스에 @EqualsAndHashCode 어노테이션 삭제

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
refactor: void 메서드에 반환값이 있도록 변경

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
refactor: 테스트 코드에 data.sql와 mokito 적용

seonghun127 committed:
Merge remote-tracking branch 'upstream/seonghun127' into seonghun127

seonghun127 committed:
docs: 기능 구현 목록 업데이트

seonghun127 committed:
feat: 댓글 추가 기능 비동기로 전환

seonghun127 committed:
core: properties 파일 변경

seonghun127 committed:
feat: 로그인 기능 비동기로 전환

seonghun127 committed:
refactor: 테스트 코드 리펙토링

1. User 도메인의 vo 클래스에 @EqualsAndHashCode 제거한다
2. 테스트 코드에 any() 메서드를 사용한다

seonghun127 committed:
refactor: UserLoginDto 변경 및 테스트 코드 추가

seonghun127 committed:
fix: UserService 이메일 중복 로직 수정

seonghun127 committed:
refactor: Error 객체 추가

seonghun127 committed:
refactor: 회원가입 기능 비동기로 전환

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
refactor: 게시글, 댓글에 delete cascade 추가

seonghun127 committed:
feat: logback-access 추가

seonghun127 committed:
refactor: ResponseDto의 필드 타입 변경

1. ArticleReponseDto의 User 필드의 타입을
UserResponseDto로 변경한다
2. CommentResponseDto의 User, Article 필드의 타입을
UserResponseDto, ArticleResponseDto로 변경한다

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
style: 코드 포맷팅

seonghun127 committed:
style: 코드 포맷팅

gracefulBrown reviewed (changes requested):
안녕하세요. 리뷰어 브라운 입니다.
기능구현 잘 하시고 테스트 코드도 잘 구현하셨네요 👍 
로깅 라이브러리 사용은 잘해주셨는데 스크립트를 이용한 자동 배포 스크립트가 빠진것 같은데 확인해주세요 :)
몇가지 코멘트 남겼으니 확인해주시구요~
궁금한 점 있으면 언제든 DM주세요 :)

seonghun127 committed:
refactor: ResponseDto 클래스에 도메인 객체를 dto 객체로 변환하는 메서드 추가

1. 도메인에 뒀던 dto 변환 메서드를 제거한다
2. 도메인이 dto에 대해 알 필요가 없다고 판단되어 도메인 객체를 dto로 변환시키는
메서드를 dto 클래스에 선언한다

seonghun127 committed:
refactor: ajax 통신을 하는 api의 반환타입을 ResponseEntity로 변경

1. ReponseEntity의 static 메서드로 반환 Http status code를 결정한다
2. Spring HATEOAS 의존을 추가하여 해당 controller에서 URI를 생성해준다
3. created 메서드로 반환 시 2번에서 생성한 URI를 headers의 location으로 설정한다

seonghun127 committed:
docs: 자동 배포 쉘 스크립트 파일 추가

gracefulBrown reviewed (approved):
피드백 반영 👍 
http 학습을 하면서 테스트 시 검증해볼 수 있는 부분을 확인해보면 좋을 것 같아요!
수고 많으셨습니다! 머지할게요~
